lib/configuration/variables/sources/instance-dependent/get-aws.js

```diff
@@ -21,10 +21,14 @@ module.exports = (serverlessInstance) => {
 
       switch (address) {
         case 'accountId': {
-          const { Account } = await serverlessInstance
-            .getProvider('aws')
-            .request('STS', 'getCallerIdentity', {}, { useCache: true });
-          return { value: Account };
+          const provider = serverlessInstance.getProvider('aws');
+          let result;
+          if (provider._v3Enabled) {
+            result = await provider.requestV3('STS', 'getCallerIdentity', {}, { useCache: true });
+          } else {
+            result = await provider.request('STS', 'getCallerIdentity', {}, { useCache: true });
+          }
```

Comment:
Why not use a feature flag for gradual rollout but can we hide this in the existing API so that all these changes become unnecessary?

Reply:
The feature flag is just temporary. The full work-plan has us remove v2 completely at the end. I think this is probably just fine. It was probably a mistake and unnecessary for plugins to use our sdk instance. They can get creds from us, but they should make their own sdk instances. They can make that change, and sill support the version before this PR and after.

Reply:
OK, We do have a feature flag. `SLS_AWS_SDK_V3=true` can be used as an env variable to force everything to use v3.

Reply:
Not all the v3 calls map perfectly onto v2 like this method, so its not possible in general to abstract, and provide a unified interface. Probably wasted work if we plan to immediately rip out v2, if we agree that is the way to go. I don't think we're going to get much feedback from people until after we break things. :trollface: